#12 Curator and zookeeper versions incompatibilities

MegaPapa opened the pull request.

andrus commented:
@MegaPapa A few questions:

* Why did we remove a dependency on `log4j-over-slf4j` and explicit dependency on Zookeeper (which we used to exclude Log4J). Is ZK now free from LOg4J dependency?
* We upgraded Curator from 2.9.1 to 2.12.0. The current version is 4.0.1 though. Should we go all the way to 4.0.1?

MegaPapa commented:
Hello, @andrus, by your questions:
1) It's my mistake, i deleted `log4j-over-slf4j` dependency and used zookeeper's.

2) Yes, we can update Curator version to 4.0.1, but i didn't do this, because by default Curator with versions 2.12.0+ use alpha (or beta) versions of zookeeper (i changed zookeeper to 2.12.0 because it's last version of Curator, which uses stable version of zookeeper). But we can combine Curator 4+ with stable versions of zookeeper (The last stable version of zookeeper it's `3.4.13` (Jul 2018)). As documentation says - we can combine Curator 4+ with stable zookeeper 3.4+, but some features from new curator will be disabled. (https://curator.apache.org/zk-compatibility.html)

I think, it will be fine to use combination of last Curator (4.0.1) and last stable version of zookeeper (3.4.13), because we will be on the edge of stable curator-zookeeper libs.
